src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -82,11 +83,42 @@ public boolean isDeleted() {
     }
 
     public List<Answer> getAnswers() {
-        return answers;
+        return answers.getAnswers();
+    }
+
+    public void validateDeletable(NsUser loginUser) throws CannotDeleteException {
+        validateOwner(loginUser);
+        answers.validateAnswerOwner(loginUser);
+    }
+
+    private void validateOwner(NsUser loginUser) throws CannotDeleteException {
+        if (!this.isOwner(loginUser)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+    }
+
+    public List<DeleteHistory> delete(long questionId) {
```

Comment:
```suggestion
    public void delete(NsUser loginUser) {
```
Answer 피드백과 같이 현재 둘로 분리되어 있는 구현과 위와 같이 삭제 권한 체크여부와  deleted 상태 변경 로직을 함께 구현하는 위 접근 방식 중 어느 접근 방식이 좋을지 고민해 보면 어떨까?